security-manager/src/main/java/org/wildfly/extension/security/manager/DeferredPermissionFactory.java

```diff
@@ -0,0 +1,98 @@
+/*
+* Copyright 2021 Red Hat, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.wildfly.extension.security.manager;
+
+import java.security.Permission;
+import org.jboss.modules.ModuleLoader;
+import org.jboss.modules.security.LoadedPermissionFactory;
+import org.jboss.modules.security.ModularPermissionFactory;
+import org.jboss.modules.security.PermissionFactory;
+import org.wildfly.extension.security.manager.logging.SecurityManagerLogger;
+import org.wildfly.security.manager.WildFlySecurityManager;
+
+/**
+ * <p>A deferred permission factory that logs a warning if the construction fails.</p>
+ *
+ * @author rmartinc
+ */
+public class DeferredPermissionFactory implements PermissionFactory {
```

Comment:
Is this only referenced from the same package?  If so it should not be public but could it be better in the Elytron project?  We would prefer the WildFly Core code to just contain the code required for management,

Reply:
Sadly it is used by two packages: org.wildfly.extension.security.manager and org.wildfly.extension.security.manager.deployment. The first one for max and min sets (standalone.xml) and the second one for the deployment set (permissions.xml).